src/components/app/MenuButtons/index.js

```diff
@@ -94,6 +108,110 @@ class MenuButtonsImpl extends React.PureComponent<Props> {
     }
   }
 
+  _deleteThisProfileOnServer = () => {
+    this.setState({
+      metaInfoPanelState: 'delete-confirmation',
+    });
+  };
+
+  _onProfileDeleted = () => {
+    this.setState({
+      metaInfoPanelState: 'profile-deleted',
+    });
+  };
+
+  _resetMetaInfoState = () => {
+    this.setState({
+      metaInfoPanelState: 'initial',
+    });
+  };
+
+  _renderUploadedProfileActions(
+    currentProfileUploadedInformation: UploadedProfileInformation
+  ) {
+    return (
+      <div className="profileInfoUploadedActions">
+        <div className="profileInfoUploadedDate">
+          <span className="profileInfoUploadedLabel">Uploaded:</span>
+          {_formatDate(currentProfileUploadedInformation.publishedDate)}
+        </div>
+        <div className="profileInfoUploadedActionsButtons">
+          <button
+            type="button"
+            className="photon-button photon-button-default photon-button-micro"
+            onClick={this._deleteThisProfileOnServer}
+            title={
+              currentProfileUploadedInformation.jwtToken === null
+                ? 'This profile cannot be deleted because we lack the authorization information.'
```

Comment:
If we have `currentProfileUploadedInformation`, but `jwtToken` is null, it means that _this_ user uploaded the profile, but this was before we stored the jwtTokens.

Now, full disclosure: I don't think this is possible for "normal" users. This is possible for _me_ because I tested this patch before the server was doing that, but I don't think we ever deployed code where we could have this state. So I'm very tempted to make jwtToken non-optional and remove a lot of code.

So I'm gonna keep this text for now because this is more correct, but I think we'll remove this if condition in the future.